include/picongpu/particles/shapes/P4S.hpp

```diff
@@ -21,135 +21,146 @@
 
 #include "picongpu/simulation_defines.hpp"
 
+#include <cstdint>
+
+
 namespace picongpu
 {
 namespace particles
 {
 namespace shapes
 {
-
-namespace shared_P4S
-{
-
-struct P4S
-{
-    static constexpr int support = 5;
-
-    HDINLINE static float_X ff_1st_radius( float_X const x )
-    {
-        /*
-         * W(x)= 115/192 - 5/8 * x^2 + 1/4 * x^4
-         *     = 115/192 + x^2 * (-5/8 + 1/4 * x^2)
-         */
-        float_X const square_x = x * x;
-        return 115._X / 192._X + square_x * (
-            -5._X / 8._X +
-            1.0_X / 4.0_X * square_x
-        );
-    }
-
-    HDINLINE static float_X ff_2nd_radius( float_X const x )
-    {
-        /*
-         * W(x)= 1/96 * (55 + 20 * x - 120 * x^2 + 80 * x^3 - 16 * x^4)
-         *     = 1/96 * (55 + 4 * x * (5 - 2 * x * (15 + 2 * x * (-5 + x))))
-         */
-        return 1._X / 96._X * (
-            55._X + 4._X * x * (
-                5._X - 2._X * x * (
-                    15._X + 2._X * x * (
-                        -5._X + x
-                    )
-                )
-            )
-        );
-    }
-
-    HDINLINE static float_X ff_3rd_radius( float_X const x )
-    {
-        /*
-         * W(x)=1/384 * (5 - 2*x)^4
-         */
-        float_X const tmp = 5._X - 2._X * x;
-        float_X const square_tmp = tmp * tmp;
-        float_X const biquadratic_tmp = square_tmp * square_tmp;
-
-        return 1._X / 384._X * biquadratic_tmp;
-    }
-};
-
-} //namespace shared_P4S
-
-/** particle assignment shape `piecewise biquadratic spline`
- */
-struct P4S : public shared_P4S::P4S
+namespace detail
 {
-    using CloudShape = picongpu::particles::shapes::PCS;
 
-    struct ChargeAssignmentOnSupport : public shared_P4S::P4S
+    struct P4S
     {
+        static constexpr int support = 5;
```

Comment:
Why is this defined as (signed) int while `assignmentFunctionOrder` is a `unit_32`? I think it would make sense to keep both as `uint`.

Reply:
and at any other shape ...

Reply:
It would, just was `int` before and I did not touch. Could change to a fixed-size type.